bamboo/common_python/tools.py

```diff
@@ -657,3 +661,158 @@ def assert_failure(return_code, expected_error, error_file_name):
         'return_code={rc}\nFailed with error different than expected.\nactual_error={ae}\nexpected_error={ee}\nSee {efn}'.format(
             rc=return_code, ae=actual_error, ee=expected_error,
             efn=error_file_name))
+
+
+def create_tests(setup_func, test_name):
+    """Create functions that can interact with PyTest.
+
+    This function creates tests that involve setting up and running an
+    LBANN experiment with the Python frontend. `setup_func` should be
+    a function that takes in the LBANN Python module and outputs
+    objects for an LBANN experiment. A test succeeds if LBANN runs and
+    exits with an exit code of 0, and fails otherwise.
+
+    PyTest detects tests by loading in a Python script and looking for
+    functions prefixed with 'test_'. After you call this function
+    within a script to generate test functions, make sure to add the
+    test functions to the script's scope. For example:
+
+        _test_funcs = tools.create_tests(setup_func, test_name)
+        for t in _test_funcs:
+            globals()[t.__name__] = t
+
+    Args:
+        setup_func (function): Sets up an LBANN experiment using the
+            Python frontend. It takes in the LBANN Python module as
+            input and returns a `(lbann.Trainer, lbann.Model,
+            lbann.reader_pb2.DataReader, lbann.Optimizer)`.
+        test_name (str): Descriptive name. Should be prefixed with
+            'test_'.
```

Comment:
Why don't we just have the function add the `test_` prefix?

Reply:
Ah ok, if we extract the test name automatically as in `_test_name = os.path.splitext(os.path.basename(current_file))[0]`, then there's no reason to have `create_tests` add `test_`.